### PR TITLE
Update user

### PR DESCRIPTION
**Pull request by Hanane8**
#Update User 

## Description
create update user Task 

## Changes Made

Create a new Razor component Update User razor
create method Update Service
Error Handling and Validation
Styling and Design Update user Page

## Related Ticket or Issue

- Fixes #ISSUE_NUMBER 
- Closes #PROJECT_NAME/ISSUE_NUMBER 

## Checklist

- [] Code adheres to the coding conventions
- [] Unit tests are added for new code
- [] Existing unit tests are updated if needed
- [] Code follows the principles of Clean Architecture
- [] Code formatting complies with the `.editorconfig` file
- [] Pull request template is filled out

### CODE REVIEW

File: BlazorApp/wwwroot/css/app.css [hunk 1]
```diff
@@ -101,3 +101,99 @@ a, .btn-link {
 code {
     color: #c02d76;
 }
+
```
Comment (BojanMirkovic) [inline BlazorApp/wwwroot/css/app.css:104]: You don't need this CSS for styling; instead, you should use Bootstrap classes. This looks like a ChatGPT suggestion and doesn't add anything special to your code.

File: Shared-Layer/ApiServices/UserServices.cs [hunk 2]
```diff
@@ -27,9 +27,9 @@ public async Task<IEnumerable<UserModel>> GetAllUsersAsync()
             return await _httpClient.GetFromJsonAsync<List<UserModel>>("api/User/GetAllUsers");
         }
 
-        public Task<UserModel> GetUserByEmailAsync(string email)
+         public async Task<UserModel> GetUserByEmailAsync(string email)
         {
-            throw new NotImplementedException();
+            return await _httpClient.GetFromJsonAsync<UserModel>($"api/User/GetUserByEmail/{email}");
```
Comment (BojanMirkovic) [inline Shared-Layer/ApiServices/UserServices.cs]: you should not change this part at this moment.You are working on UpdateUser and GetUserByEmailAsync has nothing to do with your Task.

File: BlazorApp/Pages/UpdateUser.razor [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    {
+    }
+
+
+    private async Task HandleValidSubmit()
+    {
+        try
+        {
+            await UserService.UpdateUserAsync(updatingUserModel);
+            Navigation.NavigateTo("/profile-updated");
```
Comment (BojanMirkovic) [inline BlazorApp/Pages/UpdateUser.razor]: This should be commented out until you create the profile-update component, or you should navigate somewhere else. Alternatively, you can simply display a success message to the user when the update is successful.

File: Test-Layer/Test-Layer.csproj [hunk 4]
```diff
@@ -23,6 +23,7 @@
     <PackageReference Include="NUnit.Analyzers" Version="3.6.1" />
     <PackageReference Include="coverlet.collector" Version="6.0.0" />
     <PackageReference Include="System.ComponentModel.TypeConverter" Version="4.1.0" />
+    <PackageReference Include="System.Runtime" Version="4.3.1" />
```
Comment (BojanMirkovic) [inline Test-Layer/Test-Layer.csproj:26]: why did you instal this package?

**Review by BojanMirkovic — CHANGES_REQUESTED** [review]

File: BlazorApp/wwwroot/css/app.css (hunk 1 above)
Comment (Hanane8) [reply BlazorApp/wwwroot/css/app.css:104]: Det var från början och jag glömde att ta bort den, dessutum det är inte en Chatgpt  och jag jobbar inte alls med den, det finns något bättre 😐

File: Test-Layer/Test-Layer.csproj (hunk 4 above)
Comment (Hanane8) [reply Test-Layer/Test-Layer.csproj:26]: för att jag får error om jag har inte lagt den.

**Review by BojanMirkovic — APPROVED** [review]

**Pull request merged**